src/runtime/hexagon/hexagon/hexagon_device_api_v2.cc

```diff
@@ -50,34 +50,54 @@ void HexagonDeviceAPIv2::GetAttr(Device dev, DeviceAttrKind kind, TVMRetValue* r
   }
 }
 
+// DataSpace: static allocations for Hexagon
 void* HexagonDeviceAPIv2::AllocDataSpace(Device dev, int ndim, const int64_t* shape,
                                          DLDataType dtype, Optional<String> mem_scope) {
-  return new HexagonBuffer(ndim, shape, dtype, mem_scope.defined() ? mem_scope : String("global"));
+  CHECK(TVMDeviceExtType(dev.device_type) == kDLHexagon);
+  CHECK(ndim == 1 || ndim == 2);
+
+  size_t typesize = (dtype.bits / 8) * dtype.lanes;
+  size_t nbytes = shape[0] * typesize;
```

Comment:
When the allocation is 2d `shape[0]` wil be the number of allocations to make, and `shape[1]` will be the allocation size (nbytes). Thus you'll need to conditionally calculate the number of bytes based on whether it is a 1 or 2d allocation.

Reply:
Good catch.  I forgot to clean this up / add the 2D case.